# HG changeset patch
# User Mike Miller <mtmiller@ieee.org>
# Date 1391988074 18000
#      Sun Feb 09 18:21:14 2014 -0500
# Branch stable
# Node ID 5c0b8f2da650d7c8d97aade8270ccc7fd7362dce
# Parent  5d14a08e753395139e8c380e760ca2896a60970f
Allow save() to write to the terminal with the filename '-' (bug #41505)

* load-save.cc (parse_save_options): Treat an argument of '-' as a filename
argument instead of an invalid option, restoring the ability to write to
the terminal with save().

diff --git a/libinterp/corefcn/load-save.cc b/libinterp/corefcn/load-save.cc
--- a/libinterp/corefcn/load-save.cc
+++ b/libinterp/corefcn/load-save.cc
@@ -1138,17 +1138,17 @@ parse_save_options (const string_vector 
         {
           use_zlib  = true;
         }
 #endif
       else if (argv[i] == "-struct")
         {
           retval.append (argv[i]);
         }
-      else if (argv[i][0] == '-')
+      else if (argv[i][0] == '-' && argv[i] != "-")
         {
           error ("save: Unrecognized option '%s'", argv[i].c_str ());
         }
       else
         retval.append (argv[i]);
     }
 
   if (do_double)
